compiler/src/arm_arch_full.ml

```diff
@@ -26,6 +26,9 @@ module Arm_core = struct
 
   let known_implicits = ["NF", "_nf_"; "ZF", "_zf_"; "CF", "_cf_"; "VF", "_vf_"]
 
+  let alloc_stack_need_extra sz = 
+    not (Arm_decl.is_expandable (Conv.cz_of_z sz))
```

Comment:
If this check corresponds to whether we can allocate/free stack with `SUB sp, sp, #imm` and `ADD sp, sp, #imm`, we can also allow 12-bit immediates. We should have `is_expandable || is_w12_encoding` (this corresponds to encodings T3 and T4 from the manual, e.g. A7.7.5 for ADD).

Reply:
I will do that.

Reply:
I now use is_arith_small